plugins/go/openapi_go_generator/renderer.go

```diff
@@ -0,0 +1,329 @@
+		f.Name = propertyNameForResponseCode(responseCode.Name)
+		f.JSONName = ""
+		response := responseCode.Value.GetResponse()
+		if response != nil && response.Schema != nil && response.Schema.GetSchema() != nil {
+			f.Type = "*" + typeForSchema(response.Schema.GetSchema())
+			t.Fields = append(t.Fields, &f)
+			if f.Name == "OK" {
+				m.ResultTypeName = typeForSchema(response.Schema.GetSchema())
+			}
+		}
+	}
+
+	t.Name = name
+	if len(t.Fields) > 0 {
+		renderer.Types = append(renderer.Types, t)
+		return t, err
+	} else {
+		return nil, err
+	}
+}
+
+func (renderer *ServiceRenderer) loadOperation(op *openapi.Operation, method string, path string) (err error) {
+	var m ServiceMethod
+	m.Name = strings.Title(op.OperationId)
+	m.Path = path
+	m.Method = method
+	m.HandlerName = "Handle" + m.Name
+	m.ProcessorName = "" + m.Name
+	m.ClientName = m.Name
+	m.ParametersType, err = renderer.loadServiceTypeFromParameters(m.Name+"Parameters", op.Parameters)
+	if m.ParametersType != nil {
+		m.ParametersTypeName = m.ParametersType.Name
+	}
+	m.ResponsesType, err = renderer.loadServiceTypeFromResponses(&m, m.Name+"Responses", op.Responses)
+	if m.ResponsesType != nil {
+		m.ResponsesTypeName = m.ResponsesType.Name
+	}
+	renderer.Methods = append(renderer.Methods, &m)
+	return err
+}
+
+// preprocess the types and methods of the API
+func (renderer *ServiceRenderer) loadService(document *openapi.Document) (err error) {
+	// collect service type descriptions
+	renderer.Types = make([]*ServiceType, 0)
+	for _, pair := range document.Definitions.AdditionalProperties {
+		var t ServiceType
+		t.Fields = make([]*ServiceTypeField, 0)
+		schema := pair.Value
+		for _, pair2 := range schema.Properties.AdditionalProperties {
+			var f ServiceTypeField
+			f.Name = strings.Title(pair2.Name)
+			f.Type = typeForSchema(pair2.Value)
+			f.JSONName = pair2.Name
+			t.Fields = append(t.Fields, &f)
+		}
+		t.Name = strings.Title(filteredTypeName(pair.Name))
+		if len(t.Fields) > 0 {
+			renderer.Types = append(renderer.Types, &t)
+		}
+	}
+	// collect service method descriptions
+	renderer.Methods = make([]*ServiceMethod, 0)
+	for _, pair := range document.Paths.Path {
+		v := pair.Value
+		if v.Get != nil {
+			renderer.loadOperation(v.Get, "GET", pair.Name)
+		}
+		if v.Post != nil {
+			renderer.loadOperation(v.Post, "POST", pair.Name)
+		}
+		if v.Put != nil {
+			renderer.loadOperation(v.Put, "PUT", pair.Name)
+		}
+		if v.Delete != nil {
+			renderer.loadOperation(v.Delete, "DELETE", pair.Name)
+		}
+	}
+	return err
+}
+
+func filteredTypeName(typeName string) (name string) {
+	// first take the last path segment
+	parts := strings.Split(typeName, "/")
+	name = parts[len(parts)-1]
+	// then take the last part of a dotted name
+	parts = strings.Split(name, ".")
+	name = parts[len(parts)-1]
+	return name
+}
+
+func typeForName(name string, format string) (typeName string) {
+	switch name {
+	case "integer":
+		if format == "int32" {
+			return "int32"
```

Comment:
Maybe not, but I'm expecting to have a version of this same file in Swift. It's redundant to write the same code twice in two languages, but I think that's necessary to interest the Swift community in building plugins.